lua/nvim_lsp/jdtls.lua

```diff
@@ -92,10 +92,43 @@ end
 
 local installer = make_installer()
 
+
+--- The presence of one of these files indicates a root directory.
+--
+-- We search for configuration files of the most common Java build systems. We
+-- could add more build systems, such as Make, but let's keep things simple for
+-- now.
+local root_files = {
+  'build.xml',             -- Ant
+  'pom.xml',               -- Maven
+  'build.gradle',          -- Gradle
```

Comment:
Adding `build.gradle` here as well could be problematic for multi-modules setups where you could have a structure like this:

```
/build.gradle
/module1/build.gradle
/module2/build.gradle
```

Reply:
I have been using those settings for months at work on a multi-module project. But maybe I was just lucky that none of the symbols were in any of the other modules. Also, wouldn't the other files have this same issue?

I think there needs to be a slightly more complicated version of `root_pattern` which looks for one of the target files in the current directory, then peeks up one directory higher, and only if it finds nothing above does it return the current directory. Opinions?